contracts/Raffles.cdc

```diff
@@ -32,10 +32,23 @@ pub contract Raffles {
         pub let source: @{RaffleSource}
         pub let details: Details
 
+        pub fun getSourceType(): Type {
+            return self.source.getType()
+        }
+
+        pub fun getAddressSource(): &AddressRaffleSource {
+            let source = &self.source as auth &{RaffleSource}
+            return source as! &AddressRaffleSource
+        }
```

Comment:
This cannot be added here, unfortunately. The Raffle itself should have no knowledge of the "source" it is using

What we likely have to do is permit returning an auth reference source to the owner of the raffle so that they can cast it to the original type at will